### PR TITLE
koji_promote: trap TypeError when checking help plugin config

### PR DESCRIPTION
**Pull request by vrutkovs**
This also fixes `add_help` plugin's incorrect verification of `go-md2man` binary

### CODE REVIEW

File: atomic_reactor/plugins/pre_add_help.py [hunk 1]
```diff
@@ -58,10 +58,10 @@ def run(self):
 
         try:
             check_output(go_md2man_cmd, stderr=STDOUT)
-        except CalledProcessError as e:
-            if e.returncode == 127:
-                raise RuntimeError(
+        except OSError:
```
Comment (lcarva) [inline atomic_reactor/plugins/pre_add_help.py]: Seems conflicting with: https://github.com/projectatomic/atomic-reactor/pull/631

Comment (vrutkovs) [inline atomic_reactor/plugins/pre_add_help.py]: Right, Tim has come up with a much better test for this. Fixed

**Comment by vrutkovs** [comment]
The tests are now correct and are failing :) This PR will be rebased after #631 is merged

File: tests/plugins/test_koji_promote.py [hunk 2]
```diff
@@ -1186,11 +1186,15 @@ def test_koji_promote_add_help(self, tmpdir, os_env, expect_result):
...
                                             release='1',
                                             session=session)
-        if expect_result in ['empty_result', 'pass']:
+
+        if expect_result == 'pass':
             workflow.prebuild_plugins_conf[AddHelpPlugin.key] = {
                 'help_file': 'foo.md',
             }
 
+        if expect_result == 'empty_config':
```
Comment (twaugh) [inline tests/plugins/test_koji_promote.py]: Could be `elif` but not really important.

**Pull request merged**